packages/headless/src/features/pagination/pagination-slice.ts

```diff
@@ -70,6 +74,18 @@ export const paginationReducer = createReducer(
       .addCase(executeSearch.fulfilled, (state, action) => {
         const {response} = action.payload;
         state.totalCountFiltered = response.totalCountFiltered;
+      })
+      .addCase(toggleSelectFacetValue, (state) => {
+        state.firstResult = 0;
```

Comment:
We just reset the current page back to 0 regardless of the number of results to show the most relevant results for the new filters.

Reply:
`getPaginationInitialState().firstResult` is better than to use the hardcoded 0.
You can also create a external method that all those cases can reuse.